src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/NodeStatsMetricsCollector.java

```diff
@@ -181,33 +179,23 @@ public void collectMetrics(long startTime) {
         if (indicesService == null) {
             return;
         }
-        
+
         try {
             //reach the end of current shardId list. retrieve new shard list from IndexService
             if (!currentShardsIter.hasNext()) {
                 populateCurrentShards();
             }
-            for(int i = 0; i < controller.getNodeStatsShardsPerCollection(); i++){
-                if (!currentShardsIter.hasNext()) {
-                    break;
-                }
-                IndexShard currentIndexShard = currentShardsIter.next().getValue();
-                IndexShardStats currentIndexShardStats = this.indexShardStats(indicesService, currentIndexShard, CommonStatsFlags.ALL);
-                for (ShardStats shardStats : currentIndexShardStats.getShards()) {
-                    StringBuilder value = new StringBuilder();
-
-                    value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds());
-                    //- go through the list of metrics to be collected and emit
-                    value.append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor)
-                            .append(new NodeStatsMetricsStatus(shardStats).serialize());
-
-                    saveMetricValues(value.toString(), startTime, currentIndexShardStats.getShardId().getIndexName(),
-                            String.valueOf(currentIndexShardStats.getShardId().id()));
-                }
-            }
+            // Metrics populated with a few shards per collection.
+            // The number is set via Cluster Settings API.
+            populateMetricValuesForFewShards(indicesService, startTime);
```

Comment:
Consider renaming this to imply that metric collection is happening in paginated fashion.